.github/workflows/ci.yaml

```diff
@@ -133,7 +133,16 @@ jobs:
         - name: Run tests
           run: |
             pip install -e .
-            cd test
-            python -m pip install pytest pymbolic psutil matplotlib
-            python -m pytest --durations=0 --tb=native -rxsw .
+            python -m pip install pytest pytest-cov pymbolic psutil matplotlib
 
+            # run current test coverage
+            python -m pytest --durations=0 --tb=native -rxsw --cov-report term --cov-report json:new_cov.json --cov=logpyle test/
+
+            # get main's test coverage
+            git clone https://github.com/illinois-ceesd/logpyle.git logpyle_main
+            pip uninstall logpyle -y
+            pip install -e logpyle_main/
+            python -m pytest --durations=0 --tb=native -rxsw --cov-report term --cov-report json:old_cov.json --cov=logpyle logpyle_main/test/
+
+            # ensure that coverage can only increase
+            python -c 'exec(compile(open("test/conftest.py", "rb").read(), "test/conftest.py", "exec")); assert_cov("new_cov.json", "old_cov.json")'
```

Comment:
Instead of compiling, can you do something like: `cd test; python -c 'from conftest import assert_cov; assert_cov("new_cov.json", "old_cov.json")'`?

Reply:
I had originally done that, but importing a raw file that isn't in the same folder gets kind of nasty. Because the directory test is not a module, it's not quite as clean as that nor well supported by standard Python libraries.

Reply:
Imho, both ways are kind of awkward, but I personally prefer [31194ba](https://github.com/illinois-ceesd/logpyle/pull/135/commits/31194bad4d9066be2ce1a8fd3012294821b672b4) over the exec/compile way.

Reply:
Never mind, should have thought about it more. Resolved in commit [31194ba](https://github.com/illinois-ceesd/logpyle/pull/135/commits/31194bad4d9066be2ce1a8fd3012294821b672b4).